core/org.osate.xtext.aadl2/src/org/osate/xtext/aadl2/validation/Aadl2JavaValidator.java

```diff
@@ -3487,7 +3488,8 @@ private void checkSubcomponentsOfExtendedAbstractImplementation(ComponentImpleme
 		for (ComponentCategory subcomponentCategory : categoriesOfInheritedSubcomponents) {
 			if (!acceptableSubcomponentCategories.contains(subcomponentCategory)) {
 				error(componentImplementation.getOwnedExtension(),
-						"A " + implementationCategory.getName()
+						"A " + implementationCategory != null ? implementationCategory.getName()
```

Comment:
I think this can be resolved in a better way. A few lines earlier is a bunch of if-else-ifs that perform `instanceof` checks and assign values to `implementationCategory`. The if-else-ifs should be exhaustive. I think it would be better to add a last else block and throw an `AssertionError` complaining that there was an unrecognized `ComponentImplementation`.